File: backend/app/languages.py
SUPPORTED_LANGUAGES = {
    "pt": "Português",
    "en": "Inglês",
    "es": "Espanhol",
    "fr": "Francês",
    "de": "Alemão",
    "it": "Italiano",
    "nl": "Holandês",
    "ja": "Japonês",
    "ko": "Coreano",
    "zh": "Chinês",
    "ru": "Russo",
    "ar": "Árabe",
}


def language_name(code: str) -> str:
    return SUPPORTED_LANGUAGES.get(code, code)


def is_supported_language(code: str) -> bool:
    return code in SUPPORTED_LANGUAGES

File: frontend/src/lib/languages.ts
export interface LanguageOption {
  code: string;
  name: string;
}

export const DEFAULT_LANGUAGES: LanguageOption[] = [
  { code: "pt", name: "Português" },
  { code: "en", name: "Inglês" },
  { code: "es", name: "Espanhol" },
  { code: "fr", name: "Francês" },
  { code: "de", name: "Alemão" },
  { code: "it", name: "Italiano" },
  { code: "nl", name: "Holandês" },
  { code: "ja", name: "Japonês" },
  { code: "ko", name: "Coreano" },
  { code: "zh", name: "Chinês" },
  { code: "ru", name: "Russo" },
  { code: "ar", name: "Árabe" },
];

export function languageName(code: string, languages = DEFAULT_LANGUAGES) {
  return languages.find((language) => language.code === code)?.name ?? code;
}
